runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcWindmillStreamFactory.java

```diff
@@ -265,18 +317,21 @@ public GetWorkerMetadataStream createGetWorkerMetadataStream(
         onNewWindmillEndpoints);
   }
 
-  private StreamObserverFactory newStreamObserverFactory() {
+  private StreamObserverFactory newStreamObserverFactory(boolean hasDeadline) {
     return StreamObserverFactory.direct(
-        DEFAULT_STREAM_RPC_DEADLINE_SECONDS * 2, windmillMessagesBetweenIsReadyChecks);
+        hasDeadline ? DEFAULT_DEADLINE_SECONDS : NO_DEADLINE, windmillMessagesBetweenIsReadyChecks);
   }
 
   @Override
   public void appendSummaryHtml(PrintWriter writer) {
     writer.write("Active Streams:<br>");
-    for (AbstractWindmillStream<?, ?> stream : streamRegistry) {
-      stream.appendSummaryHtml(writer);
-      writer.write("<br>");
-    }
+    streamRegistry.stream()
+        .sorted(Comparator.comparing(AbstractWindmillStream::backendWorkerToken))
+        .collect(
```

Comment:
doing the grouping since we want the print to look like:

`
backendWorkerToken1:
- GetWorkStream
- CommitWorkStream
- GetDataStream

backendWorkerToken2:
- GetWorkStream
- CommitWorkStream
- GetDataStream

Dispatcher:
- GetWorkerMetadata
`
etc.

if we don't group and keep it flat it will look like:
`
- backendWorkerToken1-GetWorkStream
- backendWorkerToken1-CommitWorkStream
- backendWorkerToken1-GetDataStream
- backendWorkerToken2-GetWorkStream
- backendWorkerToken2-CommitWorkStream
- backendWorkerToken2-GetDataStream
- Dispatcher-GetWorkerMetadata
`
etc.

I found the former nested view easier to read.

Reply:
> yea i wanted Dispatcher to show up at the end or beginning, but i its more cosmetic

removed